### PR TITLE
[Beta] Detect Distro Already Loaded in Agent

### PR DESCRIPTION
**Pull request by JacksonWeber**
Should detect if the distro package is already present in the RP file system when attempting to attach the agent.

### CODE REVIEW

File: src/agent/agentLoader.ts [hunk 1]
```diff
@@ -236,24 +236,28 @@ export class AgentLoader {
     private _sdkAlreadyExists(): boolean {
         try {
             // appInstance should either resolve to user SDK or crash. If it resolves to attach SDK, user probably modified their NODE_PATH
-            let appInstance: string;
+            let sdkAppInstance: string;
```
Comment (hectorhdzg) [inline src/agent/agentLoader.ts:239]:
```suggestion
            let shimInstance: string;
```

File: src/agent/agentLoader.ts [hunk 2]
```diff
@@ -236,24 +236,28 @@ export class AgentLoader {
...
-                appInstance = require.resolve(`${process.cwd()}/node_modules/applicationinsights`);
+                sdkAppInstance = require.resolve(`${process.cwd()}/node_modules/applicationinsights`);
+                distroAppInstance = require.resolve(`${process.cwd()}/node_modules/@azure/monitor-opentelemetry`);
             }
             // If loaded instance is in Azure machine home path do not attach the SDK, this means customer already instrumented their app
-            if (appInstance.indexOf("home") > -1) {
+            if (sdkAppInstance.indexOf("home") > -1 || distroAppInstance.indexOf("home") > -1) {
                 const diagnosticLog: IDiagnosticLog = {
-                    message: `Azure Monitor Application Insights Distro already exists. Module is already installed in this application; not re-attaching. Location: ${appInstance}`,
+                    message: `Azure Monitor Application Insights Distro already exists. Module is already installed in this application; not re-attaching. Location: ${sdkAppInstance || distroAppInstance}`,
```
Comment (hectorhdzg) [inline src/agent/agentLoader.ts:260]: It would be good to differentiate the message depending which package was detected.

Comment (JacksonWeber) [reply src/agent/agentLoader.ts:260]: I'll make that more clear.

**Review by hectorhdzg — APPROVED** [review]
LGTM

**Pull request merged**